Compilation error: Line truncation

## Description

Compiling as outlined in the README with GCC11.4:

```sh
Fyaml/tests/test_utils.F90:34:132:

   34 |     character(len=*), parameter :: TEST_MULTI_DOC_FILE = SOURCE_DIR//"/test_example_multi_doc.yaml"
      |                                                                                                                                    1
Error: Line truncated at (1) [-Werror=line-truncation]
Fyaml/tests/test_utils.F90:34:113:

   34 |     character(len=*), parameter :: TEST_MULTI_DOC_FILE = SOURCE_DIR//"/test_example_multi_doc.yaml"
      |                                                                                                                 1
Error: Unterminated character constant beginning at (1)
Fyaml/tests/test_utils.F90:1002:70:

 1002 |         write(error_unit,*) "Loading test file: "//TEST_MULTI_DOC_FILE
      |                                                                      1
Error: Symbol ‘test_multi_doc_file’ at (1) has no IMPLICIT type; did you mean ‘test_multiple_docs’?
f951: some warnings being treated as errors
make[2]: *** [tests/CMakeFiles/test_utils.dir/build.make:78: tests/CMakeFiles/test_utils.dir/test_utils.F90.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:208: tests/CMakeFiles/test_utils.dir/all] Error 2
make: *** [Makefile:91: all] Error 2
```

## Workaround

Added `-ffree-line-length-none` to CMAKE:

```diff
$ git diff
diff --git a/CMakeLists.txt b/CMakeLists.txt
index 0820ddf..260a2c3 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -19,6 +19,15 @@ cmake_print_variables(CMAKE_Fortran_COMPILER CMAKE_Fortran_COMPILER_ID CMAKE_BUI
 # Set default compiler options for Fortran
 set(_fyaml_compiler_options "")
 
+# Add a flag to ignore line-truncation errors
+if(CMAKE_Fortran_COMPILER_ID MATCHES "GNU")
+  # For free form source files. For fixed form, consider -ffixed-line-length-none.
+  list(APPEND _fyaml_compiler_options -ffree-line-length-none)
+elseif(CMAKE_Fortran_COMPILER_ID MATCHES "Intel")
+  # Intel Fortran: allow extended source lines.
+  list(APPEND _fyaml_compiler_options -extend_source)
+endif()
+
 # First ensure we're setting platform-specific flags
 if(APPLE)
     # Get MacOS SDK path
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -16,9 +16,6 @@ endif()
 
 cmake_print_variables(CMAKE_Fortran_COMPILER CMAKE_Fortran_COMPILER_ID CMAKE_BUILD_TYPE)
 
-# Set default compiler options for Fortran
-set(_fyaml_compiler_options "")
-
 # First ensure we're setting platform-specific flags
 if(APPLE)
     # Get MacOS SDK path
@@ -44,6 +41,13 @@ if(APPLE)
         "${SDKROOT}/usr/include"
         "/usr/local/include"
     )
+endif()
+
+# Set default compiler options for Fortran
+set(_fyaml_compiler_options "")
+
+if(CMAKE_Fortran_COMPILER_ID MATCHES "GNU")
+  list(APPEND _fyaml_compiler_options -ffree-line-length-none)
 endif()
 
 # Add debug flags for Fortran

